Source/Plugins/Tilemaps/Core/Tilemaps/TilemapFloodFill.cs

```diff
@@ -0,0 +1,213 @@
+﻿using System;
+
+namespace Duality.Plugins.Tilemaps
+{
+	/// <summary>
+	/// A helper class that provides a flood fill algorithm.
+	/// </summary>
+	public static class TilemapFloodFill
```

Comment:
We are dealing with static state here now due to the static activeFillBuffer field. Remove static from this class and its members and just use a instance of the TilemapFloodFill in FillTilemapTool to solve this.